File: backend/src/main/java/org/kmurygin/healthycarbs/mealplan/dto/MealPlanDTO.java
package org.kmurygin.healthycarbs.mealplan.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kmurygin.healthycarbs.user.dto.UserDTO;

import java.time.Instant;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
public class MealPlanDTO {
    private Long id;
    private UserDTO user;
    private List<MealPlanDayDTO> days;
    private double totalCalories;
    private double totalCarbs;
    private double totalProtein;
    private double totalFat;
    private Instant createdAt;
}

File: frontend/src/app/core/models/dto/mealplan.dto.ts
import type {UserDto} from './user.dto';
import type {MealPlanDayDto} from "./mealplan-day.dto";
import type {MealPlanSource} from "@core/models/enum/mealplan-source.enum";

export interface MealPlanDto {
  id: number;
  user: UserDto;
  days: MealPlanDayDto[];
  totalCalories: number;
  totalCarbs: number;
  totalProtein: number;
  totalFat: number;
  createdAt: string;
  source: MealPlanSource;
}
